### ISSUE TITLE
✨ Enhancement: Review the implementation of the JSON editor to better management of state/style

### ISSUE THREAD

**Issue by benjagm** [json-schema-org/website#560]
### Is your feature request related to a problem? Please describe

Related: #417 

As of now we are using 2 different code editors: 1 JSON and JSON Schema (json editor) and another for the rest of languages based on [react-syntax-highlighter](https://github.com/react-syntax-highlighter/react-syntax-highlighter) library.

This is providing an inconsistent experience in terms of formatting but also in terms of features. Ideally we'll use one single document that provides:
- Consistent formating/color/linting
- Availability of copy/pasting
- In cases of JSON and availability to add labels to express that is a valid JSON. See:

<img width="546" alt="Screenshot 2024-03-18 at 19 01 55" src="https://github.com/json-schema-org/website/assets/40007659/87f8d7ca-6445-4316-9b03-10b0f54cb567">

### Describe the solution you'd like

Use or extend a code editor or improve the current implementation using 2 different editors.

Use only the [react-syntax-highlighter](https://github.com/react-syntax-highlighter/react-syntax-highlighter) and explore how to better do the customizations required:
- Lang labels for JSON and JSON Schema.
- Copy paste button.
- line number
- line highlight
- For json instances availability to highlight if is a valid instance, like the previous custom instance.

For copy paste see: https://www.bojanjagetic.com/post/react-syntaxhighlighter-copy-clipboard

For line numbers and line highlight see https://www.dhiwise.com/post/crafting-beautiful-code-blocks-with-react-syntax-highlighter

### Are you working on this?

No

**Comment by Adity20** [comment]
Shall I continue working on this @benjagm as I have the idea of code components

**Comment by AdityaSingh-02** [comment]
Heyy @benjagm Can I work on this??

**Comment by akkshitgupta** [comment]
I totally agree with you @benjagm to improve or extend the code editor, however, we can go with [next-mdx-remote](https://www.npmjs.com/package/next-mdx-remote) npm package that can handle rendering of mdx, formatting of any valid language or syntax on the client side. It provides a great User and Developer experience. very nicely go with NextJs as well :)

**Comment by Adity20** [comment]
Hello @akkshitgupta I tried using that but somehow it wasn't supporting some styles which needed to be applied to the rendered code.
But can give it one more chance 👍

**Comment by benjagm** [comment]
This is a big feature to be developed. Are you sure @Adity20 ?

**Comment by Adity20** [comment]
Maybe me and @AdityaSingh-02 can work together on this one if he is okay as he also worked on one issue related to this.
So that we can complete this task a bit early

**Comment by benjagm** [comment]
It will be fine for me if at least to make an analysis and a proposal, even thought you don't implement it

**Comment by Adity20** [comment]
Okay, then I'll make a plan and will present it to you soon.

**Comment by AdityaSingh-02** [comment]
Yup we can work together on this @Adity20

**Comment by DarhkVoyd** [comment]
Just checking back in on this issue. Is anyone actively working to resolve this issue?

**Comment by Adity20** [comment]
Yes

**Comment by jayprakash25** [comment]
Is the issue resolved or can I work on this @benjagm ?

**Comment by benjagm** [comment]
Hi everyone! There has been no updates for the last 3 months and this is an important feature. Therefore i am going to make this issue available for other contributors.

Thanks a lot for your efforts!

**Comment by benjagm** [comment]
I just updated the description of the issue suggesting to focus all the solutions on using the same library we use for the standard code editor: **react-syntax-highlighter**

**Comment by jayprakash25** [comment]
Hi @benjagm . Can I start working on this issue?

**Comment by rishabhknowss** [comment]
@benjagm is anyone working on this issue ?

**Comment by 29deepanshutyagi** [comment]
hi , i want to work on this issue , kindly assign me , if it open to contribute , @benjagm

**Comment by Karan-Palan** [comment]
> I created a document where I have stated my approach [@benjagm](https://github.com/benjagm) and [@DhairyaMajmudar](https://github.com/DhairyaMajmudar) here - https://gist.github.com/Karan-Palan/073f23767c77c2040e9694345d194572
> 

@DhairyaMajmudar @benjagm , it'd be great if  I could get any feedback on this !

**Comment by benjagm** [comment]
Hi Karan. If you can find a way to unify the editors that would be great. Remember that we have another label at the bottom to express that is a valid JSON Schema instance.

Do you like to work on this?

**Comment by benjagm** [comment]
Go ahead. Good luck!

**Comment by DhairyaMajmudar** [comment]
@Karan-Palan updates on this? Do you need any help?

**Comment by Karan-Palan** [comment]
@DhairyaMajmudar , I got caught up with exams, however I've started the work and will raise a draft PR in 2-3 days and keep you updated if I need any help!

**Comment by Karan-Palan** [comment]
An update on this issue:
- Used `react-syntax-highlighter` instead of `slate` 
- Stopped re-rendering by adding the copy state inside a memo

https://github.com/user-attachments/assets/3608f9a9-f219-4636-b8e2-aca202f7a303

I am currently finding figuring out how to some highlight features from /lib and have the same colors that we had whilst using `slate` . Also some schemas in Drafts have unusual behavior, fixing that

However @benjagm , shall I still work on it and create a PR as this summer we will replace tailwind with some other component library?
